File: BMAT202P/poisson3.r
probability <- function(x, n, func) {
    P <- 0
    if(n == 0) {
        return(1)
    }
    for(i in x:0) {
        P <- P + func(i) * probability(x-i, n-1, func)
    }
    return(P)
}

L <- 2
print(probability(4, 3, function(x) {dpois(x, L)}))

File: BMAT_Prob_and_Stat/poisson3.py
def probability(x, n, func):
    P = 0
    if n == 0:
        return 1;
    for i in range(x, -1, -1):
        P += func(i) * probability(x-i, n-1, func)
    return P

L = 2
probability(3, 3, lambda x: dpois(x, L))